students/JerryH/Lesson09/mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+class Donor:
+
+    def __init__(self, first, last, donations = None):
+        self.first = first
+        self.last = last
+        self.donations = donations
+
+    @property
+    def full_name(self):
+        return "{} {}".format(self.first, self.last)
+
+    @property
+    def donor_history(self):
+        return sum(self.donations)
+
+
+class DonorBook:
+
+    def __init__(self, donors = []):
+        if donors is None:
+            self.donors = []
+        else:
+            self.donors = donors
+
+    def add_donor(self, donor):
+        if donor not in self.donors:
+            self.donors.append(donor)
+
+    def get_all_donor_names(self):
+        for donor in self.donors:
+            print("-->" + donor.full_name)
+
+    def list_all_donor_names_sorted(self):
+        all_names = ""
+        for each_donor_name in sorted(self.donors):
+            all_names += "{}\n".format(each_donor_name)
+        return(all_names)
+
+    @staticmethod
+    def donation_prompt():
+        return float(input("Please enter the donation amount:\n"))
+
+    @staticmethod
+    def donor_name_prompt():
+        return input("Send a Thank You - Please enter a full name or type \"list\""
+            "to list the current donors:\n")
+
+    def send_thank_you(self):
+        donor_name = None
+        while not donor_name:
+            donor_name = DonorBook.donor_name_prompt()
+            if donor_name.lower() == "list":
+                print(self.get_all_donor_names())
+                donor_name = None
+
+        donation = None
+        while not donation:
+            try:
+                donation = self.donation_prompt()
+            except ValueError:
+                print("Not a valid number! Please enter a valid number:\n")
+
+        # If the donnor doesn't exist in the donor dictionary - add his info
+        # Using defaultdict
+        donors[donor_name].append(donation)
+        print("Thank You Email:  Thansk for the donation!\n\n")
+
+
+    def create_report(self):
+        pass
+
+
+    def send_letters(self):
+        pass
+
+
+    def quit_program(self):
+    	print("Thanks for using my script! Bye!")
```

Comment:
you are on the right track here!

You `Donor` class is good, it should represent a single donor object just like you're doing.


Your `DonorBook` is also on the right track, but as mentioned in the course assignment:
"There should be no input function in the class that holds the data!"

So keep things directly related to your Donor DB (`DonorBook`) in the class, like creating generating thank you letter text.. but keep the user interaction code like `input` out of the class.